RefreshTests.GameServer/Tests/Levels/ScoreLeaderboardTests.cs

```diff
@@ -394,4 +395,60 @@ public void DoesntPlayLevelWithCountOnMainline()
         HttpResponseMessage message = client.PostAsync($"/lbp/play/user/{level.LevelId}?count=3", new ReadOnlyMemoryContent(Array.Empty<byte>())).Result;
         Assert.That(message.StatusCode, Is.EqualTo(BadRequest));
     }
+
+    [Test]
+    public void OvertakeNotificationsWorkProperly()
+    {
+        using TestContext context = this.GetServer();
+        const int userAmount = 10;
+
+        List<GameUser> users = Array.Empty<GameUser>().ToList();
+        for (int i = 0; i < userAmount; i++)
+        {
+            users.Add(context.CreateUser());
+        }
+        
+        GameLevel level = context.CreateLevel(users.First());
+
+        for (int i = 0; i < users.Count; i++)
+        {
+            GameUser user = users[i];
+            context.SubmitScore(i, 1, level, user, TokenGame.LittleBigPlanet2, TokenPlatform.PS3);
+
+            // Check that notification was sent to the last #1 users
+            if (i <= 0) continue;
+
+            GameUser notificationRecipient = users[i - 1];
+
+            bool notificationSent = false;
+            foreach (GameNotification notification in notificationRecipient.Notifications)
+            {
+                if (notification.Title != "Score overtaken") continue;
+                
+                notificationSent = true;
+                context.Database.DeleteNotification(notification);
+            }
+            
+            if (i - 1 == 0) // Notification should not be sent if the last #1 entry had a score of 0
+                Assert.That(notificationSent, Is.EqualTo(false));
+            else
+                Assert.That(notificationSent, Is.EqualTo(true));
+
+            // Check that notification was not sent to people who weren't previously #1
+            if (i <= 1) continue;
+            
+            notificationRecipient = users[i - 2];
+
+            notificationSent = false;
+            foreach (GameNotification notification in notificationRecipient.Notifications)
+            {
+                if (notification.Title != "Score overtaken") continue;
```

Comment:
This condition could break if the title changes. Are there other notifications being sent conflicting with your test?

Reply:
Not right now, but I figured that it prevents the test from breaking if other notifications get implemented later

Maybe this could be improved if we make a `NotificationType` enum?

Reply:
Should I assume that the overtake notification is the only one that's been sent for now then?

Reply:
Yeah.